txp/Pos/Txp/Toil/Utxo/Functions.hs

```diff
@@ -127,28 +125,18 @@ verifyConsistency inputs witnesses
     errFmt = ("length of inputs != length of witnesses "%"("%int%" != "%int%")")
     errMsg = sformat errFmt (length inputs) (length witnesses)
 
-verifyOutputs :: VTxContext -> TxAux -> VerificationRes
+verifyOutputs :: VTxContext -> TxAux -> Either ToilVerFailure ()
 verifyOutputs VTxContext {..} (TxAux UnsafeTx {..} _) =
-    verifyGeneric $
-    concatMap verifyOutput (enumerate $ toList _txOutputs)
+    mapM_ verifyOutput (enumerate $ toList _txOutputs)
   where
-    verifyOutput :: (Int, TxOut) -> [(Bool, Text)]
-    verifyOutput (i, (TxOut {txOutAddress = addr@Address {..}, ..})) =
-        [ ( not vtcVerifyAllIsKnown || areAttributesKnown addrAttributes
-          , sformat
-                ("output #"%int%" with address "%addressF%
-                 " has unknown attributes")
-                i addr
-          )
-        , ( not $ vtcVerifyAllIsKnown && isUnknownAddressType addr
-          , sformat ("output #"%int%" sends money to an address with unknown "
-                    %"type ("%addressF%"), this is prohibited") i addr
-          )
-        , ( not (isRedeemAddress addr)
-          , sformat ("output #"%int%" sends money to a redeem address ("
-                    %addressF%"), this is prohibited") i addr
-          )
-        ]
+    verifyOutput :: (Word32, TxOut) -> Either ToilVerFailure ()
+    verifyOutput (i, (TxOut {txOutAddress = addr@Address {..}, ..})) = do
+        unless (vtcVerifyAllIsKnown && not (areAttributesKnown addrAttributes)) $
```

Comment:
It should be `when (vtcVerifyAllIsKnown && not (areAttributesKnown addrAttributes))`.
If `not vtcVerifyAllIsKnown || areAttributesKnown addrAttributes` is `False`, it's an error (in the previous version of this code).
Straightforward transformation is `unless (not vtcVerifyAllIsKnown || areAttributesKnown addrAttributes) $ throwError …`. Which is equivalent to `when (vtcVerifyAllIsKnown && not (areAttributesKnown addrAttributes))` which is probably a bit more understandable.